go/database/mpt/proof.go

```diff
@@ -47,8 +47,9 @@ type WitnessProof struct {
 func CreateWitnessProof(nodeSource NodeSource, root *NodeReference, address common.Address, keys ...common.Key) (WitnessProof, error) {
 	proof := proofDb{}
 	visitor := &proofExtractionVisitor{
-		nodeSource: nodeSource,
-		proof:      proof,
+		nodeSource:        nodeSource,
+		proof:             proof,
+		updateChildHashes: nodeSource.getConfig().HashStorageLocation == HashStoredWithNode,
```

Comment:
This flag may be unnecessary, since each node tracks dirty child hashes already. Those dirty-flags are properly set by the decoder of the node, and in case of hashes being stored with the node, they should all be dirty.

So you could simply consider this flag to always to be `true` and rely on the dirty flags of the hashes in the nodes.

Reply:
ok, good idea, updated.